test: add comprehensive unit tests for validateFormData utility

gitbiruk2010 opened the pull request:
- Adds 74 test cases covering all validation rules
- Test required fields, email format, capacity validation, tag selection, and URL validation
- Includes edge cases, boundary values, and error message verification

Additional fix for activity.controller.spec.ts, an auto fixed from linting fix -- no code changes here.


<!-- Please use the following format for the title:-->
<!-- <repo>_<sprint#>_<issue#>_<PR-title> -->

<!-- Example: appdev-repo_88_888_example-pr-name -->

<!-- Please fill out the following: -->
## Summary & Changes 📃
- **Resolves:** #57 

- **Summary:** (Briefly describe what this PR does)
  - 🔨 What does this issue fix?
     - Adds unit test for validateformdata file
  - 👀 What is the expected behavior?
     - all tests should pass
  - 🗨️ Any relevant technical details?

- **Changes:**
  - ✅ List key changes made
  - 🛠️ Mention breaking changes (if any)
  - 🔗 Link relevant discussions/issues
  - 📝 Additional info to assist developers & reviewers


## Screenshots / Visual Aids 🔎
<sub><i>📌 **Required for:** UI changes, layout updates, or bug fixes.</i></sub>

<details>
  <summary> Expand ⬇️ </summary>
  <!-- add GIFs/Screenshots/Videos/Diagrams here -->

https://github.com/user-attachments/assets/b5efa932-f656-498c-b058-10a8cc474fb1


</details>


## How to Test 🧪
1. **Steps to Reproduce:**
   - Step 0: git pull origin main 
   - Step 1: Navigate to the frontend
   - Step 2: run `npm run test validateFormData.test.tsx and observe all tests pass
2. **Expected Behavior:** (Describe what should happen)
   - All 74 tests should pass in the unit
4. **Actual Behavior (if bug):** (Describe what happens instead)


## Checklist ✅

- [ ] I have **tested** this PR **locally** and it works as expected.
- [ ] This PR **resolves an issue** (`Resolves #issue-number`).
- [ ] **Reviewers, assignees(self), tags, and labels** are correctly assigned. <!-- on the menu to the right -->
- [ ] **Squash commits** and enable **auto-merge** if approved.

gitbiruk2010 committed:
test: add comprehensive unit tests for validateFormData utility

- Add 74 test cases covering all validation rules
- Test required fields, email format, capacity validation, tag selection, and URL validation
- Include edge cases, boundary values, and error message verification
- Achieve full coverage for form validation logic

Additional fix for activity.controller.spec.ts, an auto fixed from linting fix -- no code changes here.

katrina-l-hill reviewed (approved):
All 74 tests passed successfully on my end. The test coverage is comprehensive and clearly touches every required validation rule. Great job on this, Biruk!

**Screenshot**
<img width="672" height="137" alt="Screenshot 2025-11-28 105601" src="https://github.com/user-attachments/assets/45542667-fef3-463d-8221-13ab84de875d" />

beimnettes reviewed (approved):
Hy @gitbiruk2010 i tested this branch locally and ran the full unit test suite for validateFormData. All 74 tests passed successfully on my machine with no issues. 

I’ve attached screenshots showing the branch checkout and the successful test results.
<img width="508" height="91" alt="Screenshot 2025-11-28 at 4 09 28 PM" src="https://github.com/user-attachments/assets/86f344b6-d6b9-4601-8f47-924061cde512" />

gitbiruk2010 committed:
fix: add whitespace trimming to form field validations

- Trim whitespace for eventTitle, eventDescription, eventLocation, eventHost, and eventContact
- Update email validation to only check format when field has non-whitespace content
- Add whitespace validation tests for all required string fields
- Cleaned up 'no longer needed' code blockes from validateFormData.tsx

gitbiruk2010 committed:
fixed the operator precedence bug

gitbiruk2010 committed:
Merge branch 'main' into add-formValidation-unitTest

NahomAlemu reviewed (approved):
I verified that this unit test file has full coverage. Every function and major execution path is tested. The suite exercises both the success and failure paths for each validation rule, covers important edge cases, and ensures consistent behavior between Activity and ActivityDatabase. No missing branches or untested critical logic from what I can see.
<img width="623" height="132" alt="Screenshot 2025-12-02 at 9 43 34 AM" src="https://github.com/user-attachments/assets/6f6d16df-578c-4893-a759-d7e2bbe73ad5" />

gitbiruk2010 committed:
Merge branch 'main' into add-formValidation-unitTest
